code/modules/modular_computers/hardware/virus_disk.dm

```diff
@@ -0,0 +1,140 @@
+/obj/item/computer_hardware/hard_drive/role/virus
+	name = "\improper generic virus disk"
+	icon_state = "virusdisk"
+	var/charges = 5
+
+/obj/item/computer_hardware/hard_drive/role/virus/proc/send_virus(obj/item/modular_computer/tablet/target, mob/living/user)
+	return
+
+/obj/item/computer_hardware/hard_drive/role/virus/clown
+	name = "\improper H.O.N.K. disk"
+
+/obj/item/computer_hardware/hard_drive/role/virus/clown/send_virus(obj/item/modular_computer/tablet/target, mob/living/user)
+	if(charges <= 0)
+		to_chat(user, span_notice("ERROR: Out of charges."))
+		return
+
+	if(target)
+		user.show_message(span_notice("Success!"))
+		charges--
+		target.honkamnt = rand(15, 25)
+	else
+		to_chat(user, span_notice("ERROR: Could not find device."))
+
+/obj/item/computer_hardware/hard_drive/role/virus/mime
+	name = "\improper sound of silence disk"
+
+/obj/item/computer_hardware/hard_drive/role/virus/mime/send_virus(obj/item/modular_computer/tablet/target, mob/living/user)
+	if(charges <= 0)
+		to_chat(user, span_notice("ERROR: Out of charges."))
+		return
+
+	if(target)
+		user.show_message(span_notice("Success!"))
+		charges--
+
+		var/obj/item/computer_hardware/hard_drive/drive = target.all_components[MC_HDD]
+
+		for(var/datum/computer_file/program/messenger/app in drive.stored_files)
```

Comment:
I feel like using a `locate()` here would work better, you shouldn't really have more than one messenger app on your tablet, right?